server/consumer.go

```diff
@@ -634,9 +644,17 @@ func (mset *stream) addConsumerWithAssignment(config *ConsumerConfig, oname stri
 	o.ackSubj = fmt.Sprintf("%s.*.*.*.*.*", pre)
 	o.nextMsgSubj = fmt.Sprintf(JSApiRequestNextT, mn, o.name)
 
+	// If not durable determine the inactive threshold.
+	if !o.isDurable() {
+		if o.cfg.InactiveThreshold != 0 {
+			o.dthresh = o.cfg.InactiveThreshold
```

Comment:
Would it make sense to constrain minimum inactive threshold around max ack time? They should expire some time after the last ack is expected, important for WQs etc

Reply:
If this ack can be extended, that activity should also extend the inactivity timer(?)

Reply:
Pull consumer logic is different, if we have seen a request in that window we are good, if not attempt to expire all invalid requests and if empty, then we consider inactive.

Reply:
We do not follow that philosophy with ephemerals today. E.g. push consumers if the delivery subject loses interest for some period of time (5s default IIRC) we reclaim the consumer, regardless of pending acks.

This stays consistent but we could consider adding this, but I feel pretty strong we would not want to do that.

One app sends ack that gets lost and then disconnects, we will never clean up an ephemeral pull consumer if the logic requires the acks.

Reply:
Not “never” just up to max ack waiting time.

Reply:
Possibly but that would be inconsistent with how we do that today for push. So prefer to keep them the same for now and add in that logic if needed to a later date.

Also, if I am thinking about this correctly, if we only wait an additional max wait but take no other action, this feels like just reclaiming the consumer since we don't wait for non acked messages. IMO..

Reply:
Yeah - my thinking is if you are processing messages from a WQ removing the consumer too early while there are still messages in flight could result in processed messages not being removed from tbe stream. 

Agree it’s something we can park for a later time but seems important.

Reply:
Good point but if its WQ and the consumer goes away that releases all of the messages anyway that were being held for that consumer.